sub-packages/bionemo-geneformer/src/bionemo/geneformer/data/singlecell/dataset.py

```diff
@@ -320,3 +329,40 @@ def process_item(  # noqa: D417
             "loss_mask": loss_mask,
             "is_random": torch.zeros_like(masked_tokens, dtype=torch.int64),
         }
+
+
+def _profile_sc_dataset():
+    import random
+    import time
+
+    from tqdm import tqdm
+
+    from bionemo.geneformer.data.singlecell.preprocess import GeneformerPreprocess
+    from bionemo.testing.data.load import load
```

Comment:
So, right -- I don't think this function (_profile_sc_dataset) really belongs here; this seems like something that should live in `scripts` or `tests`.

I think if we do want to move `load` to `bionemo.core`, I'd just update all the calls and tests accordingly. I don't think it's super helpful to re-export it

Reply:
I think this could be moved out into its own place. But for that, it should (1) have an API and (2) be rewritten to be generic so that any dataset & preprocessor can be used and (3) have more precise timing info.

I think it's ok to keep this non-API-facing, private function & utility around a bit. We should explicitly consider this something that will be changed and thus break in the near future though!

Reply:
Yeah I mean the goal here was to put a useful script in the module if you want to do some quick iteration on trying to make the dataset run faster.

Reply:
I've merged #412 now so this can be updated to use core now @jstjohn